### PR TITLE
slightly relax batch norm check

### PR DESCRIPTION
**Pull request by bwilbertz**
As already raised in #4704, the current LayerSetup method of the BatchNormLayer expects to have **no** LayerParamter configured at all, but also adds 3 LayerParameter itself (cf. [batch_norm_layer.cpp](https://github.com/BVLC/caffe/blob/master/src/caffe/layers/batch_norm_layer.cpp#L39-L44)).
This all is done to avoid any unintenional backward modification of the global stats in the BatchNormLayer and therefore forcing `lr_mult = 0.f`.

As a consequence, a valid (upgraded) net, which was passed once into `Net<Dtype>(net_param)` and read back via `ToProto(...)` cannot be passed a second time into `Net` without being upgraded again.

This PR slightly relaxes the checks in LayerSetUp, to just ensure `lr_mult == 0.f` or adding it, if the LayerParameter is missing (as it was introduced in #4704).

### CODE REVIEW

**Comment by shelhamer** [comment]
Thanks for issuing a patch for this @bwilbertz and sorry I couldn't handle it immediately. I'll try to check and merge today.

Note that the CI failures seem to be some bogus issue with CUDA packages on the Travis machines: https://travis-ci.org/BVLC/caffe/jobs/163831672 We can hopefully resolve that shortly. @lukeyeager have you seen this issue w/ Travis?

**Comment by lukeyeager** [comment]
> @lukeyeager have you seen this issue w/ Travis?

Yeah we're working on it - sorry.

**Comment by shelhamer** [comment]
Thanks again for this @bwilbertz -- merged after checking it myself since the CI issue is unrelated.

@lukeyeager thanks for looking into the CI problem!
